### ISSUE TITLE
Please separate the issue of unused functions from SC2317 (Command appears to be unreachable)

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#2966]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2317
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

Script: https://github.com/micro5k/microg-unofficial-installer/blob/ecf1f9b/utils/device-info.sh

#### Here's what shellcheck currently says:

```
[Line 882:](javascript:setPosition(882, 3))
  adb 1> /dev/null 2>&1 shell '
  ^-- [SC2317](https://www.shellcheck.net/wiki/SC2317) (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).
```

#### Here's what I wanted or expected to see:

Function is unused.

**Edit:** Updated.

**Comment by brother** [comment]
> This bug only happens on https://www.shellcheck.net/ but not on the release.

Bug present at [d80fdfa9e8e](https://github.com/koalaman/shellcheck/commit/d80fdfa9e8e738827a88505b26d3e596c0f0e875) as well as the current HEAD (04a8624).

not sure what version "but not on the release." refers to, probably something before 0.10.0 as that falls between the two hashes above and both those show the same output locally.

**Comment by ale5000-git** [comment]
I have tried v0.10.0 on Windows and I didn't get any "info".

**Comment by ale5000-git** [comment]
So there 2 possibilities:
1) The bug doesn't happen on Windows, or
2) Dataflow Analysis is never active on Windows even when specifically asked.

**Comment by brother** [comment]
Interesting.
Did you get other info lines? (maybe the severity setting is passed somewhere?).

**Comment by ale5000-git** [comment]
Sorry, I had an rc file in my repository and didn't remember it.

Now with `--norc` I get:
Version 0.8.0 => OK
Version 0.9.0 => issue
Version 0.10.0 => issue

**Comment by koalaman** [comment]
The line does appear to be unreachable, but ShellCheck should probably still not warn about uninvoked functions.

**Comment by ale5000-git** [comment]
Just noticed now that all infos refer to the unused function.

In my opinion there should be a separate SC code (with a more clear message) for the unused functions (and just one per function).

**Comment by brother** [comment]
@ale5000-git maybe you can clean up the original report here and retitle the issue to make it to the point?

**Comment by stevecj** [comment]
I am actually seeing `Command appears to be unreachable.` for every line in a function that actually _is_ called from later in the script. The lines that call the function are reachable, and shellcheck does not claim otherwise. I am running v0.10.0.

**Comment by ale5000-git** [comment]
@stevecj
Yours is a separate issue, also you need to post an example otherwise he cannot know why it happens.

**Comment by koalaman** [comment]
Unreachable functions now have a separate SC2329, and unreachable commands within them will no longer emit SC2317. This also fixes `x=$(y)` getting two separate SC2317, one for the assignment and one for the command in the substitution. Thanks!

**Comment by robstoll** [comment]
@koalaman it looks like this change is included in version 0.10.0 in snap but not in the release on github. Would it be possible to release a 0.10.1 to be aligned again?

**Comment by brother** [comment]
The change is in git master HEAD and is not part of any official release. If you follow the guide in the README you will use the edge channel and that is built rolling. To follow the releases you need to snap install from stable/default instead.

The release will come when deemed necessary (some might even argue that nine months is well overdue though =)).

![image](https://github.com/user-attachments/assets/e4b2e3a6-3fc7-4063-813e-7191012859bd)
